scripts/src/public/services/orders.ts

```diff
@@ -109,17 +110,33 @@ async function createOrder(offer: offerDb.Offer, user: User) {
 		// replaceTemplateVars(offer, offer.meta.order_meta.content!)
 		meta: offer.meta.order_meta,
 		blockchainData: {
-			sender_address: offer.type === "spend" ? user.walletAddress : offer.blockchainData.sender_address,
-			recipient_address: offer.type === "spend" ? offer.blockchainData.recipient_address : user.walletAddress
+			sender_address: senderAddress,
+			recipient_address: recipientAddress
 		}
 	});
 
+	if (order.type === "spend" && recipientAddress) {
+		await addWatcherEndpoint([recipientAddress], order.id);
+	}
+
 	await order.save();
 	metrics.createOrder("marketplace", offer.type, offer.id);
 
 	return order;
 }
 
+async function extractMarketplaceOrderWalletAddresses(user: User, offer: offerDb.Offer) {
+	const app = await Application.findOneById(user.appId);
+	if (!app) {
+		throw NoSuchApp(user.appId);
+	}
+	const senderAddress = offer.type === "spend" ? user.walletAddress : app.walletAddresses.sender;
+	// spend marketplace offers shouldn't exists in production, and sending the funds to app wallet instead root wallet requires
+	// client change, for now keep it like before (client pay to root wallet address)
+	const recipientAddress = offer.type === "spend" ? offer.blockchainData.recipient_address : user.walletAddress;
```

Comment:
won't ```offer.blockchainData.recipient_address``` always be the root wallet? Since the root wallet is specified as the recipient for all marketplace orders

I think it should be ```app.walletAddresses.recipient```